packages/backend-common/src/reading/GcsUrlReader.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Storage } from '@google-cloud/storage';
+import {
+  ReaderFactory,
+  ReadTreeResponse,
+  SearchResponse,
+  UrlReader,
+} from './types';
+import getRawBody from 'raw-body';
+import { Logger } from 'winston';
+
+const parseURL = (
+  url: string,
+): { host: string; bucket: string; key: string } => {
+  const { host, pathname } = new URL(url);
+
+  if (host !== 'storage.cloud.google.com') {
```

Comment:
hmmm I'm a bit wary of supporting the browser url. For example the url I see is `https://console.cloud.google.com/storage/browser/_details/<bucket>/<path>?project=<proj>` instead of the one you posted here. It seems it might be more volatile than the blob url that the console displays as the "Authenticated URL". Should I add support for these two as well?  I'm happy to do the work, just double checking that this is what we want.

Reply:
The one I posted in the previous comment was just the URL from the console view of the file. 

The one this PR supports is the one from inside that view where it says "Authenticated URL":
![yaml_gcs](https://user-images.githubusercontent.com/736631/111754381-39240300-8898-11eb-8b01-a0bd0ef9e59c.png)

Reply:
yep ok, let's keep it for now!